src/test/java/dk/sdu/se_f22/sortingmodule/range/rangepublic/DoubleFilterTest.java

```diff
@@ -33,21 +45,54 @@ class useFilterMethod {
         //
         //It must also be able to take an empty list.
 
-        @Test
+        @ParameterizedTest
+        @DisplayName("Changing product attribute actually changes attribute used for filtering")
+        @ValueSource(strings = {"price", "averageUserReview", "clockspeed", "weight"})
+        void changingProductAttributeActuallyChangesBeingFiltered(String productAttribute) throws RangeFilterException {
+            //Creating filter through getTestFilter method
+            DoubleFilter filter = getTestFilter(productAttribute);
+            double userMin = 10.0;
+            double userMax = 100.0;
+
+            filter.setUserMin(userMin);
+            filter.setUserMax(userMax);
+
+
+            //Preparing input list
+            List<Product> mockResults = Helpers.readMockProductResultsFromFile("rangepublic/ProductsForDoubleFilterTest.csv", true);
+
+            // preparing the expected result list
+            // Inserting the product, that is within the range of usermin and usermax
+            ArrayList<Product> expectedResults = new ArrayList<>();
+            if (productAttribute.equals("price")) {
+                expectedResults.add(mockResults.get(0));
+            }
+            if (productAttribute.equals("averageUserReview")) {
+                expectedResults.add(mockResults.get(1));
+            }
+            if (productAttribute.equals("clockspeed")) {
+                expectedResults.add(mockResults.get(2));
+            }
+            if (productAttribute.equals("weight")) {
+                expectedResults.add(mockResults.get(3));
+            }
+
+            Collection<Product> filteredResults = filter.useFilter(mockResults);
+
+            Assertions.assertEquals(expectedResults, filteredResults);
+        }
+
         @DisplayName("filter a list of actual products")
```

Comment:
// The reason for the test fail is know, got to comment starting with here.
DOES IT STILL FAIL BABY